locations/spiders/save_on_foods_ca.py

```diff
@@ -0,0 +1,32 @@
+from scrapy import Spider
+
+from locations.items import Feature
+from locations.pipelines.address_clean_up import clean_address
+
+# SaveOnFoods
+
+
+class SaveOnFoodsCASpider(Spider):
+    name = "Save_On_Foods_ca"
+    item_attributes = {"brand": "Save-On-Foods", "brand_wikidata": "Q7427974"}
+    start_urls = ["https://storefrontgateway.saveonfoods.com/api/stores"]
+
+    def parse(self, response):
+        for store in response.json()["items"]:
+
+            item = Feature()
+            item["lat"] = store["location"]["latitude"]
+            item["lon"] = store["location"]["longitude"]
+
+            item["street_address"] = clean_address(
+                [store["addressLine1"], store["addressLine2"], store["addressLine3"]]
+            )
+            item["city"] = store["city"]
+            item["postcode"] = store["postCode"]
+            item["country"] = store["country"]
+
+            item["phone"] = store["phone"]
+            item["name"] = store["name"]
+            item["ref"] = store["retailerStoreId"]
+            # StoreHours provided by the API but not parsed by this scraper; please add
```

Comment:
Don't be afraid to go after the opening hours; though this one is a little non standard there seem to be only a few distinct patterns:
```
"openingHours":"Easter Sunday 07:00 AM-06:00 PM,
"openingHours":"Easter Sunday 07:00 AM-06:00 PM,
"openingHours":"Everyday 07:00 AM-09:00 PM",
"openingHours":"Everyday 07:00 AM-10:00 PM",
"openingHours":"Everyday 07:00 AM-10:00 PM,
"openingHours":"Everyday 07:00 AM-11:00 PM",
"openingHours":"Everyday 08:00 AM-07:00 PM",
"openingHours":"Everyday 08:00 AM-08:00 PM",
"openingHours":"Everyday 08:00 AM-09:00 PM",
"openingHours":"Everyday 08:00 AM-10:00 PM",
"openingHours":"Everyday 08:00 AM-10:00 PM,
"openingHours":"Saturday 07:00 AM-10:00 PM,
"openingHours":"Saturday 08:00 AM-08:00 PM,
"openingHours":"Saturday 08:00 AM-09:00 PM,
"openingHours":"Weekdays 07:00 AM-11:00 PM,
"openingHours":"Weekdays 08:00 AM-06:00 PM,
"openingHours":"Weekdays 08:00 AM-10:00 PM,
"openingHours":"Weekdays 09:00 AM-09:00 PM,
```

To tackle a situation like this; there's the `add_ranges_from_string` method. You might simply want to do some string replacements of "Weekdays" to "Mo-Fr" and "Everyday" to "Mo-Su" before passing it to the method:
https://github.com/alltheplaces/alltheplaces/blob/4cda3aae1f656ff6954acb22e04af96ffa1a12ee/locations/spiders/ccc.py#L45